upup/pkg/fi/cloudup/awsup/aws_cloud.go

```diff
@@ -557,6 +562,58 @@ func deleteInstance(c AWSCloud, i *cloudinstances.CloudInstance) error {
 	return nil
 }
 
+// deregisterInstanceFromClassicLoadBalancer ensures that connectionDraining completes for the associated loadBalancer to ensure no dropped connections.
+// if instance is associated with an NLB, this method no-ops.
+func deregisterInstanceFromClassicLoadBalancer(c AWSCloud, i *cloudinstances.CloudInstance) error {
+	asg := i.CloudInstanceGroup.Raw.(*autoscaling.Group)
+
+	asgDetails, err := c.Autoscaling().DescribeAutoScalingGroups(&autoscaling.DescribeAutoScalingGroupsInput{
```

Comment:
I agree that this would be ideal, but not 100% sure where the best place for this is, especially since we only need this info for rolling-updates and not any other kops operations.